apps/web/src/actions/user/setupAction.ts

```diff
@@ -16,7 +19,7 @@ export const setupAction = errorHandlingProcedure
       return z.object({
         returnTo: z.string().optional(),
         name: z.string().min(1, { message: 'Name is a required field' }),
-        email: z
+        email: z // This email validation needs to be conditional if invite_only and token is present
```

Comment:
The same as the other comment. Let's do a different 

`/inviations/:toke` page that use a new `user/acceptInvitation` action with just the logic necessary for accepting the invitation.